Sketcher/SketchUnmap.FCMacro

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/python
+#####################################
+# Copyright (c) openBrain 2019
+# Licensed under LGPL v2
+#
+# This macro will unmap a sketch from eg. a face and make its placement absolute in the body. It proposes 3 options (forcely the first if sketch not in a body) :
+# * "Raw" mode => the sketch placement is made absolute in the body referential, nothing more
+# * "DP@Face mode" => a datum plane is created where the mapping face is, then the sketch is attached to it respecting its attachment offset
+# * "DP@Sketch" mode => a datum place is created where the sketch is (including attachment offset), then the sketch is attached to its origin
+#
+# Main use of the macro is to make design more robust to topological naming issue
+#
+# Version history :
+# *0.5 : beta release
+# *0.6 : some typo improvement + commenting for official PR
+#
+#####################################
+
+    __Name__ = 'SketchUnmap'
+    __Comment__ = 'Unmap a sketch & makes its placement absolute"'
+    __Author__ = 'openBrain'
+    __Version__ = '0.6'
+    __Date__ = '2019-06-10'
+    __License__ = 'LGPL v2'
+    __Web__ = 'https://www.freecadweb.org/wiki/Macro_SketchUnmap'
+    __Wiki__ = 'https://www.freecadweb.org/wiki/Macro_SketchUnmap'
+    __Icon__ = ''
+    __Help__ = 'Select the sketch to unmap (eg. in the tree view) then run the macro'
+    __Status__ = 'Beta'
+    __Requires__ = 'FreeCAD >= 0.17'
+    __Communication__ = 'https://forum.freecadweb.org/viewtopic.php?f=22&t=36078'
+    __Files__ = ''
+
+__dbg__ = False  #True for debugging
+
+from PySide import QtGui
+
+def cslM(msg): #Print message in console
+    FreeCAD.Console.PrintMessage('\n')
+    FreeCAD.Console.PrintMessage(msg)
+
+def cslW(msg): #Print warning in console
+    FreeCAD.Console.PrintMessage('\n')
+    FreeCAD.Console.PrintWarning(msg)
+
+def cslE(msg): #Print error in console
+    FreeCAD.Console.PrintMessage('\n')
+    FreeCAD.Console.PrintError(msg)
+
+def cslD(msg): #Print debug message in console
+    if __dbg__:
+        FreeCAD.Console.PrintMessage('\n')
+        FreeCAD.Console.PrintMessage("Debug : " + str(msg))
+
+_0Vec_ = FreeCAD.Vector(0, 0, 0) #Shortener for null vector
+_ZVec_ = FreeCAD.Vector(0, 0, 1) #Shortener for Z axis
+
+if __dbg__:  ##Clear report view in debug mode
+    FreeCADGui.getMainWindow().findChild(QtGui.QTextEdit, "Report view").clear()
+
+cslM("Starting SketchUnmap macro")
+cslD("Checking selection")
+
+if (len(Gui.Selection.getSelection()) != 1 or str(Gui.Selection.getSelection()[0]) != '<Sketcher::SketchObject>'): ##If selection is wrong, print error message and exit
```

Comment:
Please put comment above the code block in this case

Reply:
Sorry but could you be more precise on what you mean by "in this case" ? In which case(s) do you expect the comment to be above the code block ?

Reply:
I think when the comment extends to far right past the margin. @galou are we planning to use `pep8` to keep coding style to a certain standard.

Reply:
IMHO, strictly applying PEP8 could refrain users to contribute. ;)